test/integration/retryable-reads/retryable_reads.spec.test.js

```diff
@@ -28,3 +37,87 @@ describe('Retryable Reads', function () {
     );
   });
 });
+
+// These tests are skipped because the driver 1) executes a ping when connecting to
+// an authenticated server and 2) command monitoring is at the connection level so
+// when the handshake fails no command started event is emitted.
+const SKIP = [
+  'find succeeds after retryable handshake network error',
+  'find succeeds after retryable handshake network error (ShutdownInProgress)'
+];
+
+describe('Retryable Reads (unified)', function () {
+  runUnifiedSuite(loadSpecTests(path.join('retryable-reads', 'unified')), SKIP);
+});
+
+describe('Retryable Reads', function () {
+  const dbName = 'retryable-handshake-tests';
+  const collName = 'coll';
+  const docs = [
+    { _id: 1, x: 11 },
+    { _id: 2, x: 22 },
+    { _id: 3, x: 33 }
+  ];
+  let client;
+  let db;
+  let coll;
+
+  beforeEach(function () {
+    if (
+      semver.lt(this.configuration.buildInfo.version, '4.2.0') ||
+      !VALID_TOPOLOGIES.includes(this.configuration.topologyType)
+    ) {
+      this.currentTest.skipReason =
+        'Retryable reads tests require MongoDB 4.2 and higher and no standalone';
+      this.skip();
+    }
+    client = this.configuration.newClient({});
+    db = client.db(dbName);
+    coll = db.collection(collName);
+  });
+
+  afterEach(async function () {
+    if (db) {
+      await db.admin().command({
+        configureFailPoint: 'failCommand',
+        mode: 'off'
+      });
+      await coll.drop();
+      await client.close();
+    }
+  });
+
+  context('when the handshake fails with a network error', function () {
+    it('retries the read', async function () {
+      await client.connect();
+      await coll.insertMany(docs);
+      await db.admin().command({
+        configureFailPoint: 'failCommand',
+        mode: { times: 2 },
+        data: {
+          failCommands: ['saslContinue', 'ping'],
+          closeConnection: true
+        }
+      });
+      const documents = await coll.find().toArray();
```

Comment:
in the original test, the find is with an `_id: 2`, in both test cases, I think we can keep that for consistency

Reply:
Updated to find one with _id of 2.